functions/server_array_to_hash.pp

```diff
@@ -0,0 +1,12 @@
+# @summary Function to normalise servers/pools/peers
+#
+# @api private
+function chrony::server_array_to_hash(Variant[Hash,Array] $servers, $options = []) >> Hash {
+  if $servers.is_a(Hash) {
+    $servers
```

Comment:
no familiar with this module but wonder if the following (or similar) may make senses
```
  $servers.map. |$server| { {$servers[0] => $servers[1] + $options
```

Reply:
`map` will always return an Array, and we specifically need to construct a hash.  If the input was already a Hash, we just return it unmodified.

Reply:
Oh yes i'm pretty sure you can do the following 
```
  Hash($servers.map. |$server| { [$servers[0], $servers[1] + $options] })
```

The intentiion here being that if a hash is passed then $options is merged e.g.

```puppet
$foo = {'foo.example.org' => ['foo']}
chrony::server_array_to_hash($foo, ['bar'])  ==  {'foo.example.org' => ['foo', 'bar']}
```
however this behaviour may not be obvious from the function name and its sounds like its undesirable so i will resolve